Newcld recognitions

dcsiebe opened the pull request:
Document the new way of performing recognitions.
This change should not be merged to master but to newcld since newcld isn't used yet.

dcsiebe committed:
docs(audio): self-describing audio formats

Document self-describing audio formats. These will be used later on.

dcsiebe committed:
docs(choice_recognitions): new audio stream

Document new way of streaming audio/performing a choice recognition.
Instead of sending the audio before performing the recognition, stream
the audio using a registered procedure and perform a streaming
recognition.

kvdb commented on docs/websocket/audio.md:
```diff
@@ -20,3 +20,17 @@ name         | type  | description
+The [SoX documentation][1] elaborates on this in more technical details.
+
+Basically, any self-describing audio format *should* work, however only WAVE is
+supported.
```
supported -> guaranteed to work

dcsiebe replied:
I feel less comfortable to say it's guaranteed, it implies next level stability imo. Any reasons for this?

kvdb commented on docs/websocket/audio.md:
```diff
@@ -20,3 +20,17 @@ name         | type  | description
+Basically, any self-describing audio format *should* work, however only WAVE is
+supported.
+
+To support streams, WAVE for example omits the length of the audio in the
```
perhaps use a heading: WAVE to explain the details about how we support wave as official format as well as a hack without the length. Perhaps a link to this 'extended standard'

dcsiebe replied:
I'll just omit this, I might be wrong here. After reading something about it (http://soundfile.sapp.org/doc/WaveFormat/) it seems that omitting the size in the header means each chunk should have a size 'header'. I think I'm adding more confusion than clarity here.

kvdb commented on docs/websocket/choice_recognitions.md:
```diff
@@ -70,30 +46,36 @@ organisation_id | `string` | **Required** The id of the organisation in which to
 ## Perform the recognition
 
-After completing the streaming of the audio, the recognition can be performed.
+The recognition can be performed when all preparations are done.
```
perhaps it makes sense to add a new heading/paragraph on the 'dance' we're expecting client/server to perform before diving into to details without explaining the overall idea. It might be easier to grok the concept:
* call initialise audio 
* register rpc, stream audio
* start to recognise audio
(this might no longer reflect current reality but you get the idea, right ;-)

dcsiebe replied:
Added it to the top of the file ;)

dcsiebe committed:
docs(audio): remove wave header note

Remove a note on wave headers which is basically out of scope.

dcsiebe committed:
docs(choice_recognitions): add summary of rpcs

Add a summary of which rpcs need to be called in order to perform a
choice recognition.
